pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -45,16 +45,57 @@ type Controller struct {
 
 // NewController creates a new Aggregate controller
 func NewController() *Controller {
+
 	return &Controller{
-		registries: make([]Registry, 0),
+		registries: []Registry{},
 	}
 }
 
 // AddRegistry adds registries into the aggregated controller
 func (c *Controller) AddRegistry(registry Registry) {
 	c.storeLock.Lock()
 	defer c.storeLock.Unlock()
-	c.registries = append(c.registries, registry)
+
+	registries := c.registries
+	registries = append(registries, registry)
+	c.registries = registries
+}
+
+// DeleteRegistry deletes specified registry from the aggregated controller
+func (c *Controller) DeleteRegistry(clusterID string) {
+	c.storeLock.Lock()
+	defer c.storeLock.Unlock()
+
+	if len(c.registries) == 0 {
+		log.Warnf("Registry list is empty, nothing to delete")
+		return
+	}
+	index, ok := c.GetRegistryIndex(clusterID)
+	if !ok {
+		log.Warnf("Registry is not found in the registries list, nothing to delete")
+		return
+	}
```

Comment:
I'd like a log here to clarify which cluster is being removed from registry

Reply:
done